### PR TITLE
Rework `etcd` chain element to distinguish between `begin.Unregister()` and `nse.Unregister()` events

### PR DESCRIPTION
**Pull request by NikitaSkrynnik**
## Issue
https://github.com/networkservicemesh/sdk-k8s/issues/456

## Description
How we distinguish between events from `begin` and NSE:
1. `etcd` puts NSE version into context on NSE register
2. `expire` uses the version from this context when it calls `begin.Unregister()`. (Changes for `expire` can be found here: https://github.com/networkservicemesh/sdk/pull/1533)
3. if `etcd` on Unregister gets the version for the context, it know that this Unregister event has been sent from `begin`. If context doensn't have the version, `etcd` just deletes NSE without version checking.

### CODE REVIEW

File: pkg/registry/etcd/nse_server.go [hunk 1]
```diff
@@ -136,6 +138,17 @@ func (n *etcdNSERegistryServer) Unregister(ctx context.Context, request *registr
...
+	if _, ok := nseVersionFromContext(ctx); !ok {
+		err = n.client.NetworkservicemeshV1().NetworkServiceEndpoints(n.ns).Delete(
+			ctx,
+			request.Name,
+			metav1.DeleteOptions{})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to delete a NetworkServiceEndpoints %s in a namespace %s", request.Name, n.ns)
+		}
+		return resp, nil
+	}
```
Comment (denis-tingaikin) [inline pkg/registry/etcd/nse_server.go:150]: Could you please say a bit more why do we need this?

**Comment by dualBreath** [comment]
The following tests failed:
`Test_K8sNSRegistry_FindWatch`
`TestScaledRegistry_NSEUnregisterInAnotherRegistryUseCase`
Could current changes cause this problem?

**Comment by NikitaSkrynnik** [comment]
@dualBreath These test will start working when this PR https://github.com/networkservicemesh/sdk/pull/1542 is merged.

**Comment by dualBreath** [comment]
LGTM

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**